public/js/reader.js

```diff
@@ -323,30 +337,22 @@ Reader.goToPage = function (page, fromHistory = false) {
     Reader.showingSinglePage = false;
     if (Reader.doublePageMode && Reader.currentPage > 0 && Reader.currentPage < Reader.maxPage) {
         // composite an image and use that as the source
-        img1 = Reader.loadImage(Reader.pages[Reader.currentPage], Reader.canvasCallback);
-        img2 = Reader.loadImage(Reader.pages[Reader.currentPage + 1], Reader.canvasCallback);
+        const img1 = Reader.loadImage(Reader.currentPage);
+        const img2 = Reader.loadImage(Reader.currentPage + 1);
+        let imagesLoaded = 0;
+        const loadHandler = () => { (imagesLoaded += 1) === 2 && Reader.drawCanvas(img1, img2); };
+        $([img1, img2]).each((_i, img) => {
+            img.onload = loadHandler;
+            // If the image is preloaded it does not trigger onload, so we have to call it manually
+            if (img.complete) { loadHandler(); }
+        });
     } else {
-        // In single view, just use the source URLs as is
-        $("#img").attr("src", Reader.pages[Reader.currentPage]);
+        const preloaded = Reader.loadImage(Reader.currentPage);
```

Comment:
We don't really need to know the image is preloaded in this portion of the code so I'd just rename the const to `img` here